### DISCUSSION TITLE
IService adds Async and Content-Types Specific interface methods

### DISCUSSION THREAD

**Discussion by mylvghb** [ServiceStack/Discuss#85] [Ideas]
ServiceStack's `IService` interface already has methods for verbs such as `IGet<T>`, `IPost<T>`, `IPut<T>`, etc. Could you add `*Async` suffix interface methods for these verbs as well as interface methods for specifying Content-Types to increase productivity?

Examples:

```csharp
public class HelloService : Service, IGet<Hello>, IGetAsync<Hello>, IGetJson<Hello>
{
    //  auto-impl interface methods with Interface
}
```

**Comment by mythz** [comment]
I've added Async IVerb interfaces in [this commit](https://github.com/ServiceStack/ServiceStack/commit/264343a00bdfe515ea02d241dfb2d299e17e4d2b). 

We wont be adding content type suffix interfaces which would cause an explosion of interfaces, but if there are Content-Types you use frequently you can add and use your own local Content-Type Verb interfaces for the same effect.

This change is available from v6.11.1+ that's now available in [Pre-Release Packages](https://docs.servicestack.net/pre-release).

**Reply by mylvghb** [reply]
Async IVerb sould be return Task type?

**Reply by mylvghb** [reply]
In your document

![image](https://github.com/ServiceStack/Discuss/assets/35945793/77b87169-1114-4a3c-9808-47fccb5d0dae)

**Reply by mylvghb** [reply]
In this commit: 
![image](https://github.com/ServiceStack/Discuss/assets/35945793/6e17a566-0c75-4eee-8642-57e38c3d43d6)

**Reply by mythz** [reply]
Ahh crap nice catch, yeah they should, updated in [this commit](https://github.com/ServiceStack/ServiceStack/commit/85facfe4e83899478ec21b086c35dbdf1e4fc63d).

I'll let you know when the pre-release packages are updated.

**Comment by mythz** [comment]
Now available in the latest v6.11.1+ pre-release packages.
